### PR TITLE
Jwoo/ios inline textrun

### PR DESCRIPTION
**Pull request by nesalang**
add text run feature to iOS

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                        textRunContent = [[NSMutableAttributedString alloc] initWithData:htmlData options:options documentAttributes:nil error:nil];
+                        // Drop newline char
+                        [textRunContent deleteCharactersInRange:NSMakeRange([textRunContent length] -1, 1)];
+                        lab.selectable = YES;
+                        lab.dataDetectorTypes = UIDataDetectorTypeLink;
+                        lab.userInteractionEnabled = YES;
+                    } else {
+                        textRunContent = [[NSMutableAttributedString alloc] initWithString:text attributes:descriptor];
+                        [textRunContent deleteCharactersInRange:NSMakeRange(0, 3)];
+                        [textRunContent deleteCharactersInRange:NSMakeRange([textRunContent length] -4, 4)];
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm]: Can you add a comment explaining what this is doing?

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm]: sure; thanks

---
In reply to: [272271678](https://github.com/Microsoft/AdaptiveCards/pull/2613#discussion_r272271678) [](ancestors = 272271678)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm [hunk 2]
```diff
@@ -111,6 +111,14 @@ - (ACOResolverIFType)getResolverIFType:(NSString *)scheme;
     return [_resolvers getResolverIFType:scheme];
 }
 
+- (UIColor *)getHighlightColor:(ACRContainerStyle)style
+                     foregroundColor:(ForegroundColor)color
+                  subtleOption:(bool)isSubtle
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: indenting looks off here

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLongPressGestureRecognizerFactory.mm [hunk 3]
```diff
@@ -14,54 +14,67 @@
...
 #import "ACOBaseActionElementPrivate.h"
+#import "ACRUILabel.h"
 
 using namespace AdaptiveCards;
 
 @implementation ACRLongPressGestureRecognizerFactory
 // instantiates a target for UILongPressGestureRecognizer object
 // and instantiate a tap gesture reconginizer with target, and return it
 // when failed, nil is returned
++ (NSObject<ACRSelectActionDelegate> *)buildTarget:(std::shared_ptr<BaseActionElement> const &)action
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLongPressGestureRecognizerFactory.mm]: >get: [](start = 47, length = 4)

nit: indenting looks off

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm (hunk 2 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: thanks

---
In reply to: [272289790](https://github.com/Microsoft/AdaptiveCards/pull/2613#discussion_r272289790) [](ancestors = 272289790)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLongPressGestureRecognizerFactory.mm (hunk 3 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLongPressGestureRecognizerFactory.mm]: o.k.

---
In reply to: [272290199](https://github.com/Microsoft/AdaptiveCards/pull/2613#discussion_r272290199) [](ancestors = 272290199)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                        textRunContent = [[NSMutableAttributedString alloc] initWithData:htmlData options:options documentAttributes:nil error:nil];
+                        lab.selectable = YES;
+                        lab.dataDetectorTypes = UIDataDetectorTypeLink;
+                        lab.userInteractionEnabled = YES;
+                    } else {
+                        textRunContent = [[NSMutableAttributedString alloc] initWithString:text attributes:descriptor];
+                        // text is preprocessed by markdown parser, and will wrapped by <p></P>
+                        // lines below remove the p tags
+                        [textRunContent deleteCharactersInRange:NSMakeRange(0, 3)];
+                        [textRunContent deleteCharactersInRange:NSMakeRange([textRunContent length] -4, 4)];
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm]: it looks like a fair bit of this is the same as is used by `ACRTextBlockRenderer`. are there opportunities to factor out common functionality?

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm]: I agree; I would like to have a new text processor  that handles text related tasks. I will create a separate task to track this.

---
In reply to: [272291859](https://github.com/Microsoft/AdaptiveCards/pull/2613#discussion_r272291859) [](ancestors = 272291859)

Comment (paulcam206) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRichTextBlockRenderer.mm]: sounds good -- thanks :)

---
In reply to: [272293732](https://github.com/Microsoft/AdaptiveCards/pull/2613#discussion_r272293732) [](ancestors = 272293732,272291859)

**Pull request merged**